packages/cfengine/build.sh

```diff
@@ -1,22 +1,63 @@
 TERMUX_PKG_HOMEPAGE=https://cfengine.com/
 TERMUX_PKG_DESCRIPTION="CFEngine is a configuration management technology."
 TERMUX_PKG_LICENSE="GPL-3.0"
-TERMUX_PKG_VERSION=3.15.2
-TERMUX_PKG_SRCURL=(https://cfengine-package-repos.s3.amazonaws.com/tarballs/cfengine-3.15.2.tar.gz
-                   https://cfengine-package-repos.s3.amazonaws.com/tarballs/cfengine-masterfiles-3.15.2.tar.gz)
-TERMUX_PKG_SHA256=(d1c570b7a0f47794a92f66e21cccdc86b8f56a7028a389780e705db41bfd3cab
-                   932d882cb6231db0cd9cfcaa8a0130e069355bc1a7307f8038ababd1320d99a8)
-TERMUX_PKG_DEPENDS="liblmdb, openssl, libandroid-glob, pcre"
+TERMUX_PKG_VERSION=3.16.1
+#TERMUX_PKG_SRCURL=(https://cfengine-package-repos.s3.amazonaws.com/tarballs/cfengine-3.16.1.tar.gz
+#                   https://cfengine-package-repos.s3.amazonaws.com/tarballs/cfengine-masterfiles-3.16.1.tar.gz)
+TERMUX_PKG_SRCURL=(https://github.com/craigcomstock/core/archive/termux.zip
+                   https://github.com/craigcomstock/masterfiles/archive/termux.zip
+                   https://github.com/craigcomstock/libntech/archive/termux.zip)
+TERMUX_PKG_SHA256=(60cb260e4ac5a1c3bbee9c51e0bc993e78a1397ea566643115a2b4f0f4885d1e
+                   f522da7f5f4f6b64a8cf37b691b6e67ef846817ce832e7dde9f5bede10423ae6
+                   2beaa4be30341e62bed3879bbc1fa95bb9dd5fb4380c82afbccee650fdf6981b)
+#TERMUX_PKG_SHA256=(d1c570b7a0f47794a92f66e21cccdc86b8f56a7028a389780e705db41bfd3cab
+#                   932d882cb6231db0cd9cfcaa8a0130e069355bc1a7307f8038ababd1320d99a8)
+TERMUX_PKG_DEPENDS="liblmdb, openssl, libandroid-glob, pcre, libyaml"
 # core doesn't work with out-of-tree builds
 TERMUX_PKG_BUILD_IN_SRC=true
-TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--with-workdir=$TERMUX_PREFIX/var/lib/cfengine --without-pam --without-selinux-policy --without-systemd-service --with-lmdb=$TERMUX_PREFIX --with-openssl=$TERMUX_PREFIX --with-pcre=$TERMUX_PREFIX"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--with-workdir=$TERMUX_PREFIX/var/lib/cfengine --without-pam --without-selinux-policy --without-systemd-service --with-lmdb=$TERMUX_PREFIX --with-openssl=$TERMUX_PREFIX --with-yaml=$TERMUX_PREFIX --with-pcre=$TERMUX_PREFIX --with-prefix=$TERMUX_PREFIX"
 
-termux_step_post_get_source() {
-  cd cfengine-masterfiles-${TERMUX_PKG_VERSION}
-  EXPLICIT_VERSION=${TERMUX_PKG_VERSION} ./configure --prefix=$TERMUX_PREFIX/var/lib/cfengine --bindir=$TERMUX_PREFIX/bin
+termux_step_post_extract_package() {
```

Comment:
termux_step_post_extract_package was renamed to termux_step_post_get_source in a recent commit so the code here is currently not run. 

Also, this hooks currently does several things that I think should be split up into more appropriate functions:

##`./autogen.sh`
This is normally run in termux_step_pre_configure

## Configures the project 
Done the same `./autogen.sh` command. If you run [`NO_CONFIGURE=1 ./autogen.sh`](https://github.com/cfengine/core/blob/fc7188b8097a7bc2140e70772393cb20b5e4de41/autogen.sh#L74) then the configuring can be done in termux_step_configure instead, you could maybe then also remove `TERMUX_PKG_BUILD_IN_SRC=true`, unless the error happens in the make step. However, if you want `-prefix=$TERMUX_PREFIX/var/lib/cfengine` then I guess you need to override termux_step_configure in any case.

## `make install`
`make install` should (and is in this case) be run in `termux_step_make_install`. It would be run twice right now if you would run it in termux_step_post_get_source.

Reply:
Yes, the prefix must be changed as we default to /var/cfengine which is invalid for termux environment. :)

Reply:
The autogen.sh and make here are for masterfiles and not core, so we are combining two separate repos which require autogen/make steps so not run twice on the same repo. Should be OK.